compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1045,10 +1045,7 @@ impl<'a> Context<'a> {
             }
         };
         // Ensure that array id is fully resolved.
-        let array = dfg.resolve(array);
-
-        let array_id = dfg.resolve(array);
-        let array_typ = dfg.type_of_value(array_id);
```

Comment:
Why were these lines removed? It seems we are not resolving `array` anymore, unless I'm seeing things wrong.

Reply:
Good spot! I may have been over eager there. This was the original:

```rust
        // Ensure that array id is fully resolved.
        let array = dfg.resolve(array);
        let array_id = dfg.resolve(array);       // NOTE: This is the same as `array`
        let array_typ = dfg.type_of_value(array_id);    // NOTE: For some reason in my mind I thought `type_of_value` would return the same thing, but I'm not actually sure why I thought that, I guess it could be different between the original and the resolved one
        ...
        if self.handle_constant_index_wrapper(instruction, dfg, array, index, store_value)? {. // NOTE: self.handle_constant_index_wrapper resolves the array again
            return Ok(());
        }

      // NOTE: There are two more uses that I think I completely ignored. 
```

I'll back the resolve, but this is why it would be good if methods could say if they want resolved or not.